src/Workspaces/Core/Portable/Experiments/IExperimentationService.cs

```diff
@@ -3,25 +3,64 @@
 using System.Composition;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Experiments
 {
     internal interface IExperimentationService : IWorkspaceService
     {
-        bool IsExperimentEnabled(string experimentName);
+        bool IsExperimentEnabled(ExperimentName experimentName);
     }
 
     [ExportWorkspaceService(typeof(IExperimentationService)), Shared]
     internal class DefaultExperimentationService : IExperimentationService
     {
-        public bool IsExperimentEnabled(string experimentName) => false;
+        public bool IsExperimentEnabled(ExperimentName experimentName) => false;
     }
 
     internal static class WellKnownExperimentNames
     {
-        public const string RoslynOOP64bit = nameof(RoslynOOP64bit);
-        public const string CompletionAPI = nameof(CompletionAPI);
-        public const string PartialLoadMode = "Roslyn.PartialLoadMode";
-        public const string RoslynToggleBlockComment = "Roslyn.ToggleBlockComment";
+        public readonly static ExperimentName RoslynOOP64bit = nameof(RoslynOOP64bit);
+        public readonly static ExperimentName CompletionAPI = nameof(CompletionAPI);
+        public readonly static ExperimentName PartialLoadMode = nameof(PartialLoadMode);
+        public readonly static ExperimentName ToggleBlockComment = nameof(ToggleBlockComment);
+    }
+
+    internal struct ExperimentName
+    {
+        public const string DefaultPrefix = "Roslyn.";
+
+        /// <summary>
+        /// experiement name
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// experiment name with a prefix
+        /// 
+        /// we need this since some experimentation service requires certain prefix in its name.
+        /// </summary>
+        public string NameWithPrefix { get; }
```

Comment:
I think it would make it a little easier to understand if these were renamed slightly to match the actual service they're being passed to.

NameWithprefix makes it sound like it's the same as Name but with a prefix (but you could potentially pass anything in the constructor).

How about `FeatureFlagName` and `FlightName`.  And you can enforce the dot for `FeatureFlagName` like is done now.